convert.py

```diff
@@ -0,0 +1,213 @@
+import re, collections, textwrap, sys, argparse, platform
+
+
+Field = collections.namedtuple('Field', ['name', 'contents'])
+
+Header = collections.namedtuple('Header', ['module'])
+
+Function = collections.namedtuple('Function',
+        ['pos', 'name', 'purpose', 'inputs', 'returns'])
+
+Class = collections.namedtuple('Class', ['name', 'purpose'])
+
+
+def warn(message):
+    """ Print a labelled message to stderr.  """
+    sys.stderr.write('Warning: %s\n' % message)
+
+
+def header_from_block(block):
+    """ Create a Header structure from a parsed Block.  """
+    return Header(block.fields['Module'])
+
+
+def function_from_block(block):
+    """ Create a Function structure from a parsed Block.  """
+    return Function(block.pos, block.fields.get('Function', None),
+            block.fields.get('Purpose', None), block.fields.get('Inputs', None),
+            block.fields.get('Outputs', None))
+
+
+def class_from_block(block):
+    """ Create a Class structure from a parsed Block.  """
+    return Class(block.fields.get('Class', None),
+            block.fields.get('Purpose', None))
+
+
+def parse_fields(block_contents, width):
+    """ Extract the named fields of an old-style comment block.  """
+
+    field_re = re.compile(
+            r'(?:\n *(Purpose):(.*))|(?:\n *([a-zA-Z0-9]+?):\n?(.*?)?^$)',
```

Comment:
The idea is that most fields extend until the next empty line, but in a few files there are very long 'Purpose' fields with multiple paragraphs. Therefore, we want 'Purpose' fields to consume everything until the end of the block, because we know that 'Purpose' will always be the final field.